srml/council/src/seats.rs

```diff
@@ -305,7 +305,7 @@ decl_module! {
 			<TermDuration<T>>::put(count);
 		}
 
-		fn on_finalize(n: T::BlockNumber) {
+		fn on_initialize(n: T::BlockNumber) {
 			if let Err(e) = Self::end_block(n) {
```

Comment:
We should at least rename this function as well.

Reply:
Oh wow, I actually meant to change it to
```diff
- if let Err(e) = Self::end_block(n) {
+ if let Err(e) = Self::end_block(n - 1.into()) {
```

I've added a test that I _thought_ would fail, but it doesn't because the tests simulate time passing by calling `end_block` manually at the end of the block. I've opened this issue with some of my thoughts https://github.com/paritytech/substrate/issues/2818 .